packages/vscode-docker-registries/src/clients/GenericRegistryV2/GenericRegistryV2DataProvider.ts

```diff
@@ -137,4 +151,40 @@ export class GenericRegistryV2DataProvider extends RegistryV2DataProvider {
         this.authenticationProviders.delete(registryUriString);
         // TODO: check if the map of auth providers is empty, if so, remove the root from the tree
     }
+
+    public async deleteTag(item: CommonTag): Promise<void> {
+        const digest = await this.getImageDigest(item);
+        const registry = item.parent.parent as unknown as GenericV2Registry;
+        await registryV2Request({
+            authenticationProvider: this.getAuthenticationProvider(registry),
+            method: 'DELETE',
+            registryUri: registry.baseUrl,
+            path: ['v2', item.parent.label, 'manifests', digest],
+            scopes: [`repository:${item.parent.label}:pull`],
+            headers: {
+                'accept': 'application/vnd.docker.distribution.manifest.v2+json'
+            }
+        });
+    }
+
+    public async getImageDigest(item: CommonTag): Promise<string> {

```

Comment:
This and deleteTag can probably be in RegistryV2DataProvider (the base class)

Reply:
that reminds me, I should look at if that protocol would work with GitHub, theoretically, it would right?

Reply:
GitHub is fully v2 compliant, or at least, it's supposed to be.